docs/design/gateway/lwM2M-gateway.md

```diff
@@ -0,0 +1,176 @@
+# LwM2M Gateway design
+
+## Why need LwM2M Gateway
+
+For telemetryService only support push data to the server, but for LwM2M protocol, it support both push and pull data from the device. it is hard to implement the pull data feature in the telemetryService.
```

Comment:
make clear statement for "server", do you mean the telemetry data server? and the data direction be consist when using push/pull, make it always from device to the data server. a much clear statement FYI:
Telemetry service which serve push data from device to the data server, it didn't have the feature to pull data from device the to the data server, while LwM2M normally needs to pull data from the device to the data server. in order to support this, a LwM2M Gateway is required to do this job.